src/rlic/_boundaries.py

```diff
@@ -0,0 +1,39 @@
+from typing import NewType, TypeAlias, TypedDict
+
+Bound = NewType("Bound", str)
+BoundPair = tuple[Bound, Bound]
+AnyBound: TypeAlias = Bound | BoundPair
+
+
+class AbstractBounds(TypedDict):
+    udir: AnyBound
```

Comment:
I'm now thinking I can't avoid naming running variable to describe these axes, so might as well go with x and y, for simplicity